pkg/cluster/delete.go

```diff
@@ -272,37 +283,39 @@ func (m *manager) Delete(ctx context.Context) error {
 		return err
 	}
 
-	m.log.Printf("deleting role assignments")
-	err = m.deleteRoleAssignments(ctx)
-	if err != nil {
-		return err
-	}
+	// only delete if managedByARO
+	if rgManagedByARO {
+		m.log.Printf("deleting role assignments")
+		err = m.deleteRoleAssignments(ctx)
+		if err != nil {
+			return err
+		}
 
-	m.log.Printf("deleting role definition")
-	err = m.deleteRoleDefinition(ctx)
-	if err != nil {
-		return err
-	}
+		m.log.Printf("deleting role definition")
+		err = m.deleteRoleDefinition(ctx)
+		if err != nil {
+			return err
+		}
```

Comment:
We should attempt to delete these either way

Reply:
I moved those delete roleassignment/definitions above `if rgManagedByARO` and should now always run.